### PR TITLE
refactor(v2): clean up AerosolSpecies and Lognorm classes

### LINKED ISSUE #34: refactor(v2): clean up AerosolSpecies and Lognorm classes
## Context

`AerosolSpecies` and `Lognorm` are the core data-model classes for describing aerosol populations. They are functionally correct but rough around the edges: missing `__repr__`/`__str__`, inconsistent naming, and some structural clutter. This PR cleans them up without changing their public interface.

## AerosolSpecies tasks

- [ ] Keep as a traditional class (not a dataclass) to preserve the existing call signature
- [ ] Add a `__repr__` that shows species name, kappa, and distribution summary
- [ ] Add a `__str__` that produces a human-readable one-liner (used in console output)
- [ ] Audit `__init__` for unnecessary branches, dead code, and unclear variable names; simplify aggressively
- [ ] Ensure numpydoc docstrings on `__init__` (Parameters, Raises) are complete and accurate
- [ ] Write unit tests covering construction, repr, str, and edge cases (zero kappa, single-bin)

## Lognorm tasks

- [ ] Audit and complete numpydoc docstrings on all public methods
- [ ] Add `__repr__` if not present
- [ ] Write or extend unit tests for `pdf`, `cdf`, `invcdf`, and `moment`

## Notes

- No changes to the public call signatures — this is a clean-up, not a redesign
- `__eq__` and `__hash__` are explicitly *not* needed and should not be added

### PR DESCRIPTION
**Pull request by darothen**
Closes #34.

## Summary

**`pyrcel/aerosol.py` — `AerosolSpecies`**
- Extract shared `_discretize_dist()` helper; eliminates the near-identical `Lognorm` and `MultiModeLognorm` branches in `__init__`
- Add `__str__` (one-liner: `"(NH4)2SO4: kappa=0.600, N=300.0 cm⁻³, nr=50"`); improve `__repr__` to include kappa and the distribution repr
- Fix `stats()` bug: was returning `self.rho` instead of `stats_dict`
- Remove unused `self.bins` attribute; clean up variable names (`lr` → `log_r_min`, etc.)
- Complete numpydoc docstrings (Parameters, Attributes, Raises, Examples)

**`pyrcel/distributions.py` — `Lognorm`**
- Fix `invcdf` validation: `np.any(y) < 0` → `np.any(y < 0)` (was always `False`)
- Fix `mean` formula: `mu * exp(0.5*sigma**2)` → `mu * exp(0.5*ln(sigma)**2)` (correct lognormal mean)
- Remove dead `self.base` / `self.log` attributes
- Correct `Attributes` docstring for `median` and `mean`

**`tests/test_aerosol.py` (new)**
- 23 tests: `AerosolSpecies` construction (all three distribution types), SI unit storage, bins array passthrough, zero kappa, unknown distribution, `repr`/`str`
- `Lognorm`: `pdf`, `cdf`, `invcdf` roundtrip, `invcdf` validation, zeroth/first moments, integration to N, Hypothesis property-based pdf/cdf tests

## Test plan

- [x] `uv run pytest tests/ -m "not slow"` → 137 passed (114 existing + 23 new)
- [x] `uvx ruff check pyrcel/aerosol.py pyrcel/distributions.py tests/test_aerosol.py` → clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Bug fixes in Lognorm mean and AerosolSpecies.stats() can change numerical results for parcel inputs; scope is localized to aerosol/discretization with strong new test coverage.
> 
> **Overview**
> Refactors **`AerosolSpecies`** discretization by pulling shared binning into **`_discretize_dist()`**, so **`Lognorm`** and **`MultiModeLognorm`** no longer duplicate the same logic. **`stats()`** now returns the computed stats dict instead of **`self.rho`**, and dict-only distributions get a clear error instead of silent misuse. **`__repr__`** / **`__str__`** are improved; docstrings are tightened.
> 
> **`Lognorm`** fixes: **`invcdf`** validation uses **`np.any(y < 0)`** (the old check was always false), and **`mean`** uses **`exp(0.5 * ln(sigma)²)`** instead of **`exp(0.5 * sigma²)`**. Dead **`base`** / **`log`** attributes are removed.
> 
> Adds **`tests/test_aerosol.py`** with construction, units, **`repr`/`str`**, and **`Lognorm`** pdf/cdf/invcdf/moment tests plus Hypothesis property checks.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2c9ba93c402a1bb0d85dbb65c371857a59eaf023. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by darothen** [comment]
@cursor review once

File: pyrcel/distributions.py [hunk 1]
```diff
@@ -107,7 +106,7 @@ def invcdf(self, y):
 
         """
 
-        if (np.any(y) < 0) or (np.any(y) > 1):
```
[the remark this answers is not in the document]
Comment (darothen) [reply pyrcel/distributions.py:110]: Good catch — the validation was indeed wrong. `cdf(x)` returns values in `[0, N]`, not `[0, 1]`, so the `y > 1` guard broke `invcdf(cdf(x))` for any distribution with `N > 1`. Fixed in e729779 by comparing against `self.N` and adding a roundtrip test with `N=300` to prevent regression.
